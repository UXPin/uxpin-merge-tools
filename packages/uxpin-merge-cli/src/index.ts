import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
