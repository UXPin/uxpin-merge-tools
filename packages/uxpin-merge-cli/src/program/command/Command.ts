export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
