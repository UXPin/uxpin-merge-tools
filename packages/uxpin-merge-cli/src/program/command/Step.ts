import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
