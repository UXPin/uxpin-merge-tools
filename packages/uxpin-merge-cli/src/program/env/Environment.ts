export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
