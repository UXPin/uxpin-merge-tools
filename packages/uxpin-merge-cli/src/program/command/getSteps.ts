import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
