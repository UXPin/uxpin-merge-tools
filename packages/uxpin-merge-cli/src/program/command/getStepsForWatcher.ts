import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
