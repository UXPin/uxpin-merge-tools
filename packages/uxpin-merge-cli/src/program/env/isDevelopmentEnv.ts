import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
