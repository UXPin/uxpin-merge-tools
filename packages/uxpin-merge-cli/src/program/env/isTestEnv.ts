import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
