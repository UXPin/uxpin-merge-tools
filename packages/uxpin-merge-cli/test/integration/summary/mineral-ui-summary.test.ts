import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { mineralUiSummaryStub } from '../../resources/stubs/mineralUi';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT = 60000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('summary command integration', () => {
  const { getTlsPort } = setupStubbyServer(mineralUiSummaryStub);

  describe('summary command prints ', () => {
    it('prints the list of components found in mineral-ui example', async () => {
      // when
      const output: string = await runUXPinMergeCommand({
        cwd: 'resources/repos/mineral-ui',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [Command.SUMMARY, '--webpack-config "./webpack.config.js"'],
      });

      // then
      expect(output).toMatchSnapshot();
    });
  });
});
