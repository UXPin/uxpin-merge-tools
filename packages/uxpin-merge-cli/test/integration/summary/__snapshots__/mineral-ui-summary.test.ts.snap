// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary command integration summary command prints  prints the list of components found in mineral-ui example 1`] = `
"[1mGeneral[22m

    [1mAvatar[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mButton[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mButtonGroup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCheckbox[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCheckboxGroup[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mDropdown[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mFormField[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mFormFieldDivider[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mFormFieldset[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mLink[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mMenu[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mMenuDivider[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMenuGroup[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mMenuItem[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mPopover[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mRadio[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mRadioGroup[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mPagination[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mSelect[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mText[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTable[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTabs[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTab[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTextArea[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTextInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mTooltip[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

[1mCard[22m

    [1mCard[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardActions[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardBlock[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardDivider[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCardFooter[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardImage[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardStatus[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardTitle[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mCardTitleMenu[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

[1mIcons[22m

    [1mIcon[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconArrowDropdownDown[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconArrowDropdownUp[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconCheckBoxCheck[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconCheckBoxIndeterminate[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconChevronLeft[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconChevronRight[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconClose[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconDanger[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconDangerSimple[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconExpandLess[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconExpandMore[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconMoreHoriz[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconRadioButtonCheck[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconSuccess[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconSuccessSimple[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconWarning[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconWarningSimple[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

[1mLayout[22m

    [1mBox[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mFlex[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mFlexItem[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mGrid[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mGridItem[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

    [1mStartEnd[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [32mâœ”[39m

[33mwarning [39mCannot compute default value for property \`triggerTitle\`

SyntaxError: Unexpected token ':'

 in:
[34msrc/library/Card/CardFooter/CardFooter.js[39m

"
`;
